testsuite/standalone/src/test/java/org/wildfly/extras/creaper/commands/deployments/DeployUndeployCmdTest.java

```diff
@@ -0,0 +1,118 @@
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.concurrent.TimeoutException;
+
+/**
+ * This class tests deploy and undeploy commands in standalone mode.
+ */
+@Category(WildFlyTests.class)
+@RunWith(Arquillian.class)
+@RunAsClient
+public class DeployUndeployCmdTest {
+
+    private static final String DEPLOYMENT_NAME = "test-deployment.war";
+    private static final Address TEST_DEPLOYMENT_ADDRESS = Address.of("deployment", DEPLOYMENT_NAME);
+    private static final WebArchive TEST_DEPLOYMENT = ShrinkWrap.create(WebArchive.class, DEPLOYMENT_NAME)
+            .add(new StringAsset("Hello Creaper"), "index.html");
+
+    @Rule
+    public final TemporaryFolder tmp = new TemporaryFolder();
+
+    private OnlineManagementClient client;
+    private Operations ops;
+    private Administration admin;
+
+    @Before
+    public void prepare() throws IOException, OperationException, TimeoutException, InterruptedException {
+        client = ManagementClient.online(OnlineOptions.standalone().localDefault().build());
+        ops = new Operations(client);
+        admin = new Administration(client);
+        ops.removeIfExists(TEST_DEPLOYMENT_ADDRESS);
+        admin.reloadIfRequired();
+    }
+
+    @After
+    public void cleanAndClose() throws Exception {
+        try {
+            ops.removeIfExists(TEST_DEPLOYMENT_ADDRESS);
+            admin.reloadIfRequired();
+        } finally {
+            client.close();
+        }
+    }
+
+    @Test
+    public void deployAsInputStream_commandSucceeds() throws Exception {
+        InputStream inputStream = TEST_DEPLOYMENT.as(ZipExporter.class).exportAsInputStream();
+        client.apply(new Deploy.DeployCmdBuilder(inputStream, DEPLOYMENT_NAME, true).build());
+        Assert.assertTrue(ops.exists(TEST_DEPLOYMENT_ADDRESS));
+        undeploy(DEPLOYMENT_NAME);
+    }
+
+    @Test
+    public void deployAsFile_commandSucceeds() throws Exception {
+        deployAsFile(TEST_DEPLOYMENT);
+        undeploy(TEST_DEPLOYMENT.getName());
+    }
+
+    @Test
+    public void deployAfterUndeployTest() throws Exception {
+        deployAsFile(TEST_DEPLOYMENT);
+        undeploy(TEST_DEPLOYMENT.getName());
+        deployAsFile(TEST_DEPLOYMENT);
+        undeploy(TEST_DEPLOYMENT.getName());
+    }
+
+    @Test
+    public void undeployKeepContentTest() throws OperationException, IOException, CommandFailedException {
+        deployAsFile(TEST_DEPLOYMENT);
+        client.apply(new Undeploy.Builder(TEST_DEPLOYMENT.getName()).keepContent().build());
+        assertDeploymentExists(TEST_DEPLOYMENT.getName(), true);
+        ops.removeIfExists(TEST_DEPLOYMENT_ADDRESS);
+    }
+
+    private void assertDeploymentExists(String deploymentName, boolean shouldExist)
+            throws IOException, OperationException {
+        Assert.assertEquals("Deployment exists", shouldExist, ops.exists(Address.of("deployment", deploymentName)));
```

Comment:
I believe that the message would look like: "Deployments exists, expected ...., actual ....". Maybe I've got confused by how the messages are created with TestNG vs JUnit. If it would be as I expect, I believe my original message should be fine. Will check it.